src/main/java/uk/gov/hmcts/reform/wataskconfigurationapi/ccdmapping/variableextractors/AutoAssignTaskToCaseworker.java

```diff
@@ -0,0 +1,93 @@
+import uk.gov.hmcts.reform.wataskconfigurationapi.thirdparty.camunda.CamundaValue;
+import uk.gov.hmcts.reform.wataskconfigurationapi.thirdparty.camunda.TaskResponse;
+import uk.gov.hmcts.reform.wataskconfigurationapi.thirdparty.idam.IdamSystemTokenGenerator;
+import uk.gov.hmcts.reform.wataskconfigurationapi.thirdparty.roleassignment.RoleAssignmentClient;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+@Order
+public class AutoAssignTaskToCaseworker implements TaskVariableExtractor {
+
+    private final RoleAssignmentClient roleAssignmentClient;
+    private final CamundaClient camundaClient;
+    private final AuthTokenGenerator ccdServiceAuthTokenGenerator;
+    private final AuthTokenGenerator camundaServiceAuthTokenGenerator;
+    private final IdamSystemTokenGenerator idamSystemTokenGenerator;
+
+    private static final Logger LOG = LoggerFactory.getLogger(AutoAssignTaskToCaseworker.class);
+
+    public AutoAssignTaskToCaseworker(RoleAssignmentClient roleAssignmentClient,
+                                      CamundaClient camundaClient,
+                                      @Qualifier("ccdServiceAuthTokenGenerator")
+                                          AuthTokenGenerator ccdServiceAuthTokenGenerator,
+                                      @Qualifier("camundaServiceAuthTokenGenerator")
+                                          AuthTokenGenerator camundaServiceAuthTokenGenerator,
+                                      IdamSystemTokenGenerator idamSystemTokenGenerator) {
+        this.roleAssignmentClient = roleAssignmentClient;
+        this.camundaClient = camundaClient;
+        this.ccdServiceAuthTokenGenerator = ccdServiceAuthTokenGenerator;
+        this.camundaServiceAuthTokenGenerator = camundaServiceAuthTokenGenerator;
+        this.idamSystemTokenGenerator = idamSystemTokenGenerator;
+    }
+
+    @Override
+    public Map<String, Object> getValues(TaskResponse task, Map<String, CamundaValue<Object>> processVariables) {
+        String ccdId = (String) processVariables.get(ConfigureTaskService.CCD_ID_PROCESS_VARIABLE_KEY).getValue();
+
+        List<RoleAssignment> roleAssignmentList = roleAssignmentClient.queryRoleAssignments(
+            idamSystemTokenGenerator.generate(),
+            ccdServiceAuthTokenGenerator.generate(),
+            buildQueryRequest(ccdId)
+        );
+        LOG.debug("Role assignments: {}", roleAssignmentList);
+
+        return updateTaskStateAndSetAssignee(task, roleAssignmentList);
+    }
+
+    private Map<String, Object> updateTaskStateAndSetAssignee(TaskResponse task,
+                                                              List<RoleAssignment> roleAssignmentList) {
+        Map<String, Object> taskVariables = new ConcurrentHashMap<>();
+        if (roleAssignmentList.isEmpty()) {
+            taskVariables.put("taskState", "Unassigned");
```

Comment:
I believe the convention is lower cased I'm not sure why, we can probably change it but in task management we have enums so might be worth having a different ticket to clean this up.

https://github.com/hmcts/wa-task-management-api/blob/422a1c5a5e9d32cd0d4640929448e47d6cf6371f/src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/domain/entities/camunda/TaskState.java

Reply:
done